pkg/configs/api/api.go

```diff
@@ -93,12 +97,22 @@ func (a *API) getConfig(w http.ResponseWriter, r *http.Request) {
 		return
 	}
 
-	w.Header().Set("Content-Type", "application/json")
-	if err := json.NewEncoder(w).Encode(cfg); err != nil {
+	switch ParseConfigFormat(r.Header.Get("Accept"), FormatJSON) {
+	case FormatJSON:
+		w.Header().Set("Content-Type", "application/json")
+		err = json.NewEncoder(w).Encode(cfg)
+	case FormatYAML:
+		w.Header().Set("Content-Type", "application/yaml")
+		err = yaml.NewEncoder(w).Encode(cfg)
+	default:
+		// should not be here
+		level.Error(logger).Log("msg", "unknown Format")
+		http.Error(w, err.Error(), http.StatusInternalServerError)
```

Comment:
This should be a `StatusBadRequest` error, not `StatusInternalServerError `, because it's caused by an invalid input from client.

Reply:
actually, this block is unreachable because parseConfigFormat has a default value.